### DISCUSSION TITLE
FirebaseMessaging without GoogleService-Info.plist

### DISCUSSION THREAD

**Discussion by car5ten** [firebase/firebase-ios-sdk#11961] [Q&A]
Hello,
I'm currently test if one can remove the GoogleService-Info.plist from my project and configure it via code. I was able to configure `FirebaseApp` as desired, but for `Messaging` I'm unable to do so.

Debugging the code I found that `FIRMessagingFirebaseAppID` is using the `defaultOptions`. Can I somehow change that behavior?

**Comment by paulb777** [comment]
Hi @car5ten. It's currently to use other apps than the default app for Firebase Messaging. Please upvote the open feature request - https://github.com/firebase/firebase-ios-sdk/issues/8968
